mcs/class/System/Mono.Net.Security/MonoTlsProviderFactory.cs

```diff
@@ -238,56 +237,43 @@ static void PopulateProviders ()
 			var appleTlsEntry = new Tuple<Guid,String> (AppleTlsId, typeof (Mono.AppleTls.AppleTlsProvider).FullName);
 
 			providerRegistration.Add ("default", appleTlsEntry);
+			providerRegistration.Add ("legacy", appleTlsEntry);
 			providerRegistration.Add ("apple", appleTlsEntry);
 		}
 #elif MONODROID
 		// TODO: Should be redundant		
 		static void PopulateProviders ()
 		{
-			var legacyEntry = new Tuple<Guid,String> (LegacyId, typeof (Mono.Net.Security.LegacyTlsProvider).FullName);
-
-			providerRegistration.Add ("legacy", legacyEntry);
-
-	#if MONO_FEATURE_BTLS
+#if MONO_FEATURE_BTLS
 			var btlsEntry = new Tuple<Guid,String> (BtlsId, typeof (Mono.Btls.MonoBtlsProvider).FullName);
-			if (btlsEntry != null)
-				providerRegistration.Add ("default", btlsEntry);
-			else
-	#endif
-			providerRegistration.Add ("default", legacyEntry);
+			providerRegistration.Add ("default", btlsEntry);
+			providerRegistration.Add ("legacy", btlsEntry);
+			providerRegistration.Add ("btls", btlsEntry);
+#endif
 		}
 #else
 		static void PopulateProviders ()
 		{
+			Tuple<Guid,String> appleTlsEntry = null;
+			Tuple<Guid,String> btlsEntry = null;
+
 #if MONO_FEATURE_APPLETLS
-			var appleTlsEntry = new Tuple<Guid,String> (AppleTlsId, typeof (Mono.AppleTls.AppleTlsProvider).FullName);
+			appleTlsEntry = new Tuple<Guid,String> (AppleTlsId, typeof (Mono.AppleTls.AppleTlsProvider).FullName);
+			providerRegistration.Add ("apple", appleTlsEntry);
 #endif
-			var legacyEntry = new Tuple<Guid,String> (LegacyId, typeof (Mono.Net.Security.LegacyTlsProvider).FullName);
-			providerRegistration.Add ("legacy", legacyEntry);
 
-			Tuple<Guid,String> btlsEntry = null;
 #if MONO_FEATURE_BTLS
 			if (IsBtlsSupported ()) {
 				btlsEntry = new Tuple<Guid,String> (BtlsId, typeof (Mono.Btls.MonoBtlsProvider).FullName);
 				providerRegistration.Add ("btls", btlsEntry);
 			}
 #endif
 
-#if MONO_FEATURE_APPLETLS
-			if (Platform.IsMacOS)
+			var defaultEntry = appleTlsEntry ?? btlsEntry;
+			if (defaultEntry != null) {
 				providerRegistration.Add ("default", appleTlsEntry);
```

Comment:
shouldn't this use `defaultEntry` instead of `appleTlsEntry`?

Reply:
Yeah, thanks a lot for spotting this.